### PR TITLE
LHCbDataset uses full path for LocalFile

### LINKED ISSUE #874: LocalFile inputdata
It looks as if LocalFiles cannot be used for inputdata for local testing. The directory path gets stripped off so only the namePattern is used. Is this a deliberate choice or something we should fix?

### PR DESCRIPTION
**Pull request by mesmith75**
Fixes #874 . Changed the optionsString of the LHCbDataset to use the accessURL rather than namePattern for non-DiracFile s

### CODE REVIEW

**Comment by mesmith75** [comment]
Any objections to merging?

File: python/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py [hunk 1]
```diff
@@ -300,6 +300,7 @@ def getCatalog(self, site=''):
         return xml_catalog
 
     def optionsString(self, file=None):
+        print 'Options STRING!'
```
Comment (rob-c) [inline python/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py]: This is a typo surely?

File: python/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py [hunk 2]
```diff
@@ -339,8 +340,8 @@ def optionsString(self, file=None):
                 sdatasetsnew += """ \"LFN:%s\",""" % f.lfn
                 sdatasetsold += """ \"DATAFILE='LFN:%s' %s\",""" % (f.lfn, dtype_str)
             else:
-                sdatasetsnew += """ \"PFN:%s\",""" % f.namePattern
-                sdatasetsold += """ \"DATAFILE='PFN:%s' %s\",""" % (f.namePattern, dtype_str)
+                sdatasetsnew += """ \"%s\",""" % f.accessURL()[0]
+                sdatasetsold += """ \"DATAFILE='%s' %s\",""" % (f.accessURL()[0], dtype_str)
```
Comment (rob-c) [inline python/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py]: Does the `accessURL` work for all file types? (I think I remember finishing this but I'm not sure I ever tested it)

File: python/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py (hunk 1 above)
Comment (mesmith75) [inline python/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py]: yes

File: python/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py (hunk 2 above)
Comment (mesmith75) [inline python/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py]: It does for LocalFile. I'll check for MassStorageFile as well.

Comment (mesmith75) [inline python/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py]: Yes, accessURL works for MassStorageFile - I can't think of any others for LHCb.

**Comment by mesmith75** [comment]
Any objections to merging?

**Pull request merged**